pkg/server/plugin/upstreamauthority/spire/spire.go

```diff
@@ -162,35 +172,48 @@ func (m *Plugin) MintX509CA(request *upstreamauthorityv0.MintX509CARequest, stre
 	}
 }
 
-func (m *Plugin) PublishJWTKey(req *upstreamauthorityv0.PublishJWTKeyRequest, stream upstreamauthorityv0.UpstreamAuthority_PublishJWTKeyServer) error {
-	err := m.subscribeToPolling(stream.Context())
+func (p *Plugin) PublishJWTKeyAndSubscribe(req *upstreamauthorityv1.PublishJWTKeyRequest, stream upstreamauthorityv1.UpstreamAuthority_PublishJWTKeyAndSubscribeServer) error {
+	err := p.subscribeToPolling(stream.Context())
 	if err != nil {
 		return err
 	}
-	defer m.unsubscribeToPolling()
+	defer p.unsubscribeToPolling()
 
 	// Publish JWT authority
-	resp, err := m.serverClient.publishJWTAuthority(stream.Context(), req.JwtKey)
+	resp, err := p.serverClient.publishJWTAuthority(stream.Context(), &types.JWTKey{
```

Comment:
How would you feel about adding a `ToAPIFromPluginProto` method to a new apitypes.go in pkg/common/coretypes/jwtkey that does this conversion?